runtime/include/localeModels/gpu/chpl-locale-model.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _chpl_locale_model_h_
+#define _chpl_locale_model_h_
+
+#include "sys_basic.h"
+#include "chpltypes.h"
+
+//
+// This is the type of a global locale ID.
+//
+typedef struct {
```

Comment:
The gpu locale model has sublocales, so this file needs to look like the one for either the apu or numa locale model (take your pick).  At present it looks like the one for the flat locale model, which won't work because it has no sublocale support.